src/test/java/org/broadinstitute/hellbender/utils/codecs/table/TableCodecUnitTest.java

```diff
@@ -0,0 +1,204 @@
+package org.broadinstitute.hellbender.utils.codecs.table;
+
+import htsjdk.tribble.readers.LineIterator;
+import htsjdk.tribble.readers.LineIteratorImpl;
+import htsjdk.tribble.readers.LineReader;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+
+public final class TableCodecUnitTest extends BaseTest {
+
+    @DataProvider(name = "badNames")
+    public Object[][] badNames() {
+        List<Object[]> params = new ArrayList<>();
+        params.add(new String[]{"a.tsv"});
+        params.add(new String[]{"a.table.gz"});
+        params.add(new String[]{"a.bed"});
+        params.add(new String[]{"a.bcf"});
+        params.add(new String[]{"a.hapmap"});
+        params.add(new String[]{"a.refseq"});
+        params.add(new String[]{"a.beagle"});
+        return params.toArray(new Object[][]{});
+    }
+
+    @Test(dataProvider = "badNames")
+    public void testBadNames(String badName){
+        TableCodec tc = new TableCodec();
+        Assert.assertFalse(tc.canDecode(badName), badName);
+    }
+
+    @DataProvider(name = "goodNames")
+    public Object[][] goodNames() {
+        List<Object[]> params = new ArrayList<>();
+        params.add(new String[]{"a.table"});
+        return params.toArray(new Object[][]{});
+    }
+
+    @Test(dataProvider = "goodNames")
+    public void testGoodNames(String goodName){
+        TableCodec tc = new TableCodec();
+        Assert.assertTrue(tc.canDecode(goodName), goodName);
+    }
+
+    @Test
+    public void testChrs(){
+        TableCodec tc = new TableCodec();
+        Assert.assertNotNull(tc.decode("1:1  1   2   3").getChr(), "1");
+        Assert.assertNotNull(tc.decode("chr1:1  1   2   3").getChr(), "chr1");
+        Assert.assertNotNull(tc.decode("1:1  1   2   3").toString(), "1");
```

Comment:
Assert equality with the expected contig values, not merely that they aren't null.